test/functional/wallet_migration.py

```diff
@@ -445,6 +445,39 @@ def test_no_privkeys(self):
         # After migrating, the "keypool" is empty
         assert_raises_rpc_error(-4, "Error: This wallet has no available keys", watchonly1.getnewaddress)
 
+        # Migrating a watch-only blank empty wallet (with no pubkeys)
+        self.log.info("Test migration of a pure watchonly blank empty wallet (with no pubkeys)")
+        watchonly2 = self.create_legacy_wallet("watchonly2", disable_private_keys=True, blank=True)
+        # Before migrating
+        info = watchonly2.getwalletinfo()
+        assert_equal(info["descriptors"], False)
+        assert_equal(info["private_keys_enabled"], False)
+        assert_equal(info["blank"], True)
+
+        _, watchonly2 = self.migrate_and_get_rpc("watchonly2")
+        # After migrating
+        info = watchonly2.getwalletinfo()
+        assert_equal(info["descriptors"], True)
+        assert_equal(info["private_keys_enabled"], False)
+        assert_equal(info["blank"], True)
+        self.assert_is_sqlite("watchonly2")
+
+        # Migrating a watch-only non-blank empty wallet (with no pubkeys)
+        self.log.info("Test migration of a pure watchonly non-blank empty wallet (with no pubkeys)")
+        watchonly3 = self.create_legacy_wallet("watchonly3", disable_private_keys=True, blank=False)
+        # Before migrating
+        info = watchonly3.getwalletinfo()
+        assert_equal(info["descriptors"], False)
+        assert_equal(info["private_keys_enabled"], False)
+        assert_equal(info["blank"], False)
+
+        _, watchonly3 = self.migrate_and_get_rpc("watchonly3")
+        info = watchonly3.getwalletinfo()
+        assert_equal(info["descriptors"], True)
+        assert_equal(info["private_keys_enabled"], False)
+        assert_equal(info["blank"], False)
+        self.assert_is_sqlite("watchonly3")
+
```

Comment:
Could probably re-write this test in just a few lines:
```python3
self.log.info("Test migration of a watch-only empty wallet")
for is_blank in [True, False]:
    self.create_legacy_wallet("watchonly_empty", disable_private_keys=True, blank=is_blank)
    _, watchonly_empty = self.migrate_and_get_rpc("watchonly_empty")
    info = watchonly_empty.getwalletinfo()
    assert_equal(info["private_keys_enabled"], False)
    assert_equal(info["blank"], is_blank)
```

Reply:
Yup, much better! Cheers!